velox/common/base/PeriodicStatsReportDaemon.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <folly/experimental/ThreadedRepeatingFunctionRunner.h>
+#include <map>
+#include "velox/common/caching/AsyncDataCache.h"
+#include "velox/common/caching/SsdFile.h"
+#include "velox/common/memory/MemoryArbitrator.h"
+
+namespace folly {
+class CPUThreadPoolExecutor;
+}
+
+namespace facebook::velox {
+
+namespace memory {
+class MemoryAllocator;
+}
+
+namespace cache {
+class AsyncDataCache;
+}
+
+/// Manages a background daemon thread to report stats through 'StatsReporter'.
+class PeriodicStatsReportDaemon {
+ public:
+  PeriodicStatsReportDaemon(
+      const velox::memory::MemoryAllocator* const memoryAllocator,
+      const velox::cache::AsyncDataCache* const asyncDataCache,
+      const velox::memory::MemoryArbitrator* const arbitrator);
+
+  /// Invoked to start the report daemon in background.
+  void start();
+
+  /// Invoked to stop the report daemon in background.
+  void stop();
+
+  /// Testing API used to report only once for all report methods.
+  void testingReportOnce();
+
+ private:
+  // Add a task to run periodically.
+  template <typename TFunc>
+  void addTask(TFunc&& func, size_t periodMicros, const std::string& taskName) {
+    scheduler_.add(
+        taskName,
+        [taskName,
+         periodMicros,
+         func = std::forward<TFunc>(func)]() mutable noexcept {
+          try {
+            func();
+          } catch (const std::exception& e) {
+            LOG(ERROR) << "Error running periodic task " << taskName << ": "
+                       << e.what();
+          }
+          return std::chrono::milliseconds(periodMicros / 1000);
```

Comment:
If this is always in milliseconds, then how about just pass ms? like intervalMs?